docs/design/proposals/late-initialization/late-initialization.md

```diff
@@ -0,0 +1,277 @@
+with newly calculated value which represents the latest state of the resource in AWS. AWS APIs return these default or
+updated values with Create/Update API output and Read/List/Describe API output after resource creation.
+
+There are two main problems arise due to this behavior of AWS APIs,
+
+a) When an AWS API does not strictly use the property value as is in the backend, but updates it, the desired spec of
+the K8s resource in the “etcd” does not correctly represent the actual state of resource in AWS. This is a problem because
+for K8s user, the spec of custom resource is the source of truth for state of K8s resource.
+
+b) When a default field is populated during AWS resource creation, then during reconciliation, there will always be delta
+present between desired and latest K8s representation of the custom resource and an Update AWS API call be made during
+reconciliation. This update call may or may not have any side-effects on AWS resource and currently service teams either
+place custom code to avoid this OR let the calls happen if it does not alter state of the resource.
+
+> Problem (b) can be viewed as a variation of Problem (a), where server changes the property value from “null” to something.
+This is worth noting because solving Problem (a) in a generic and consistent manner will provide the solution for
+Problem (b) as well.
+
+
+## Solutions
+
+To compare the solutions for above problem, we will use the following facets:
+
+1. Consistency with K8s conventions
+2. Consistency across ACK service controllers
+3. GitOps compatibility
+4. User experience for custom resource lifecycle 
+
+✅ → doable with no/some efforts
+
+❌ → not achievable
+
+### Solution 1
+
+In this solution, the status fields will include all the fields which are CreateOutput shape members. Current status 
+fields are the CreateOutput shape members which are not present in CreateInput shape (spec fields).
+
+During reconciliation, the Read(One|Many),Update request inputs will be created from desired state but giving precedence
+to status fields over spec in case of conflict.
+
+
+1. *Consistency with K8s conventions* ❌ 
+    1. In K8s convention, the spec fields are not duplicated into status fields. This will confuse k8s users when they 
+    describe their custom resources.
+2. *Consistency across ACK service controllers* ✅
+    1. There should be no complexity in following this approach across all ACK service controllers.
+3. *GitOps compatibility* ✅
+    1. There should be no complexity in implementing GitOps for ACK custom resources. 
+    2. Approach mentioned in low level design below should work.
+4. *User experience for custom resource lifecycle* ✅
+    1. Applying updates to custom resource can become confusing for the user.
+    2. To update custom resource spec, K8s user will have to check resource status because status will have most updated
+     value of all the fields.
```

Comment:
This is seen as `doable`? These subpoints seem like detractors that make this solution impossible.

Reply:
By the :heavy_check_mark: , I meant it is possible to have an implementation like this, but we are certainly not going to implement because of subpoints.